module/os/linux/zfs/zvol_os.c

```diff
@@ -1165,19 +1512,36 @@ int
 zvol_init(void)
 {
 	int error;
-	int threads = MIN(MAX(zvol_threads, 1), 1024);
+
+	if (zvol_threads == 0) {
+		zvol_actual_threads = num_online_cpus();
```

Comment:
Should we make this MAX(32, num_online_cpus()) for backward compatibility on smaller systems and scale benefit as we go to larger systems?